### PR TITLE
Preferentially pull the the images of critical pod

### PR DESCRIPTION
**Pull request by saberrey**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
 
/kind feature

**What this PR does / why we need it**:
This pr implements a method to preferentially pull images of critical pods based on serial pull strategy( --serialize-image-pulls=true ). To mark a Pod as critical, we can set priorityClassName for that Pod to system-cluster-critical or system-node-critical，and we can also prefer to pull the image of the pod in the specified namespaces. 
When a node fails and pods are scheduled to other nodes, we hope that images of critical pods will be pulled first to ensure that important services can started first.

**Which issue(s) this PR fixes**:
#75890
**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Review by odinuge — COMMENTED** [review]
Hey @saberrey! Even tho this looks like a nice way of solving the issue, I do think this change will require a [KEP](https://github.com/kubernetes/enhancements/tree/master/keps). It should at least be discussed within sig-node, and then a KEP makes it a lot easier and quicker to get consensus about how things should work! You can also write a post on #sig-node on slack and/or join the sig-node meetings. :smile: 

Adding a hold for now.
/hold

**Comment by pacoxu** [comment]
@saberrey if you have no time to work on the KEP, I would like to create one for this. I think this is a nice improvement for cases with --serialize-image-pulls=true especially during node failures. 

- storage pod may start before those pods which would use pv 
- kube-system pods pull image first
 
______________________________________________

BTW, --serialize-image-pulls default is  true. 
```
Pull images one at a time. We recommend *not* changing the default value on nodes that run docker daemon with version < 1.9 or an `aufs` storage backend. Issue #10959 has more details. (DEPRECATED: This parameter should be set via the config file specified by the Kubelet's `--config` flag. See https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/ for more information.)
```
If --serialize-image-pulls=true, a big image pull will make other pods that is scheduled to the same host,  pending for image pulling. If the image is pull from somewhere slow， other pod may start after minutes or even a hour.
- Big image is not good practice, but may effect others.
- docker 1.9 is really old. 

Why not set --serialize-image-pulls to false in the future release ?

File: pkg/kubelet/images/puller.go [hunk 1]
```diff
@@ -70,26 +74,78 @@ func newSerialImagePuller(imageService kubecontainer.ImageService) imagePuller {
...
 }
 
 func (sip *serialImagePuller) processImagePullRequests() {
-	for pullRequest := range sip.pullRequests {
-		imageRef, err := sip.imageService.PullImage(pullRequest.spec, pullRequest.pullSecrets, pullRequest.podSandboxConfig)
-		pullRequest.pullChan <- pullResult{
-			imageRef: imageRef,
-			err:      err,
+	// Set environment variables to identify the namespace that needs to be pulled first.
+	priorityNamespace := env.GetEnvAsStringOrFallback("PRIORITY_NAMESPACES", "kube-system")
```
Comment (pacoxu) [inline pkg/kubelet/images/puller.go:95]: priorityNamespace would be another feature in this pr.

`NamespaceSystem is the system namespace` 
You may use NamespaceSystem here.

**Pull request closed without merging**